flash/vision/backbones.py

```diff
@@ -34,129 +38,105 @@
 RESNET_MODELS = ["resnet18", "resnet34", "resnet50", "resnet101", "resnet152", "resnext50_32x4d", "resnext101_32x8d"]
 DENSENET_MODELS = ["densenet121", "densenet169", "densenet161"]
 TORCHVISION_MODELS = MOBILENET_MODELS + VGG_MODELS + RESNET_MODELS + DENSENET_MODELS
-
 BOLTS_MODELS = ["simclr-imagenet", "swav-imagenet"]
 
 
-def backbone_and_num_features(
-    model_name: str,
-    fpn: bool = False,
-    pretrained: bool = True,
-    trainable_backbone_layers: int = 3,
-    **kwargs
-) -> Tuple[nn.Module, int]:
-    """
-    Args:
-        model_name: backbone supported by `torchvision` and `bolts`
-        fpn: If True, creates a Feature Pyramind Network on top of Resnet based CNNs.
-        pretrained: if true, returns a model with backbone pre-trained on Imagenet
-        trainable_backbone_layers: number of trainable resnet layers starting from final block.
-
-    >>> backbone_and_num_features('mobilenet_v2')  # doctest: +ELLIPSIS
-    (Sequential(...), 1280)
-    >>> backbone_and_num_features('resnet50', fpn=True)  # doctest: +ELLIPSIS
-    (BackboneWithFPN(...), 256)
-    >>> backbone_and_num_features('swav-imagenet')  # doctest: +ELLIPSIS
-    (Sequential(...), 2048)
-    """
-    if fpn:
-        if model_name in RESNET_MODELS:
+@BACKBONES_REGISTRY.register_function(name="simclr-imagenet", namespace="vision", package="bolts")
+def load_simclr_imagenet(
+    path_or_url: str = f"{ROOT_S3_BUCKET}/simclr/bolts_simclr_imagenet/simclr_imagenet.ckpt", **__
+):
+    simclr = SimCLR.load_from_checkpoint(path_or_url, strict=False)
```

Comment:
```suggestion
    simclr: nn.Module = SimCLR.load_from_checkpoint(path_or_url, strict=False)
```

Reply:
If it supports load_from_checkpoint this has to be a LightningModule (even stronger typing)

Reply:
Done !